### ISSUE TITLE
pulumi not honoring dependencies during destroy for elasticache

### ISSUE THREAD

**Issue by cihansahin** [pulumi/pulumi#14840]
### What happened?

I ran into delete failure issue during "pulumi destroy". It is trying to delete a subnet group resource first which is used by my elasticache cluster even though I have subnet group specified as a dependent resource on the cluster.

### Example

My subnetgroup:

`elasticache_subnet_group = aws.elasticache.SubnetGroup(
    f"{prefix}-elasticache-subnet-group",
    name=f"{prefix}-elasticache-subnet-group",
    subnet_ids=[private_subnet_a, private_subnet_b],
    tags={"Name": f"{prefix}-elasticache-subnet-group"},
    opts=pulumi.ResourceOptions(depends_on=[private_subnet_a, private_subnet_b])
)`

and my elasticache replica group:

`elasticache_replicationgroup = aws.elasticache.ReplicationGroup(
    f"{prefix}-redis-rg",
    node_type=redis_node_type,
    description=f"{prefix}-redis-rg",
    num_cache_clusters=redis_num_cache_clusters,
    subnet_group_name=elasticache_subnet_group.name,
    security_group_ids=[elasticache_security_group.id],
    tags={"Name": f"{prefix}-redis-rg"},
    transit_encryption_enabled=True,
    at_rest_encryption_enabled=True,
    engine="redis",
    engine_version="7.1",
    port=6379,
    automatic_failover_enabled=True,
    multi_az_enabled=redis_multi_az,
    preferred_cache_cluster_azs=[
        region.name + "a", 
        region.name + "b"
    ],
    parameter_group_name="default.redis7.cluster.on",
    user_group_ids=[elasticache_usergroup.id],
    opts=pulumi.ResourceOptions(
        depends_on=[elasticache_subnet_group, elasticache_security_group, elasticache_usergroup]
    )
)`

### Output of `pulumi about`

`CLI          
Version      3.96.1
Go Version   go1.21.4
Go Compiler  gc

Host     
OS       darwin
Version  14.1.1
Arch     arm64`

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by justinvp** [comment]
@cihansahin, thanks for opening the issue and sorry for the trouble! Just want to make sure I understand correctly: `elasticache_subnet_group` is being deleted before `elasticache_replicationgroup` during `pulumi destroy`?

What are the versions of the `pulumi` and `pulumi_aws` PyPi packages (if you run `pulumi about` from within your project directory, it should list these)?

**Comment by cihansahin** [comment]
Yes. elasticache_subnet_group was being deleted before elasticache_replicationgroup during pulumi destroy. But it is failing as it's used by elasticache_replicationgroup.

I already posted "pulumi about" output in the ticket. Sorry I did not keep the project in that state to run it again.

**Comment by cihansahin** [comment]
> elasticache_subnet_group is being deleted before elasticache_replicationgroup during pulumi destro

That is correct. But it is failing as it's used by elasticache_replicationgroup.

I already posted "pulumi about" output in the ticket. Sorry I did not keep the project in that state to run it again.

**Comment by justinvp** [comment]
> I already posted "pulumi about" output in the ticket. Sorry I did not keep the project in that state to run it again.

Do you recall what the versions of the `pulumi` and `pulumi_aws` PyPi packages were? `venv/bin/python -m pip list`

**Comment by lunaris** [comment]
It looks like this has gone stale without a reproduction, so I'm going to close. Please feel free to reopen if you are still encountering issues that we can identify and fix though 🙏
